src/gui/ogr/qgsvectorlayersaveasdialog.cpp

```diff
@@ -671,11 +674,41 @@ void QgsVectorLayerSaveAsDialog::mFormatComboBox_currentIndexChanged( int idx )
   }
 }
 
+void QgsVectorLayerSaveAsDialog::mUseAliasesForExportedName_stateChanged( int state )
+{
+  mAttributeTableItemChangedSlotEnabled = false;
```

Comment:
should you ask for confirmation if it will erase some input or it's not worth it here?

Reply:
Or change this to two buttons instead so it's more obvious that clicking them will replace the existing typed values

Reply:
I think a confirmation dialog is less "gui intrusive" and enough for the typical use case